devito/compiler.py

```diff
@@ -152,12 +154,33 @@ def __init__(self, *args, **kwargs):
             log("WARNING: Running on Intel KNL without OpenMP is highly discouraged")
 
 
+class CustomCompiler(Compiler):
+    """Custom compiler based on standard environment flags
+
+    :param openmp: Boolean indicating if openmp is enabled. False by default
+
+    Note: Currently honours CC, CFLAGS, LD and LDFLAGS, with defaults similar
+    to the default GNU settings.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(CustomCompiler, self).__init__(*args, **kwargs)
+        self.cc = environ.get('CC', 'g++')
```

Comment:
By convention CC and CFLAGS are used for the C compiler, while CXX and CXXFLAGS are used for C++.